http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -254,6 +255,34 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       )
     } yield ()
 
+  override def updateDateAccessed(userInfo: UserInfo, workspaceId: WorkspaceId, runtimeName: RuntimeName)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      runtime <- RuntimeServiceDbQueries.getActiveRuntime(workspaceId, runtimeName).transaction
+
+      // If user is creator of the runtime, they should definitely be able to see the runtime.
+      hasPermission <-
+        if (runtime.auditInfo.creator == userInfo.userEmail) F.pure(true)
+        else {
+          checkSamPermission(
+            WsmResourceSamResourceId(WsmControlledResourceId(UUID.fromString(runtime.samResource.resourceId))),
+            userInfo,
+            WsmResourceAction.Write
+          ).map(_._1)
+        }
+
+      _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done auth call for update date accessed runtime permission")))
+      _ <- F
+        .raiseError[Unit](
+          RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied", Some(ctx.traceId))
+        )
+        .whenA(!hasPermission)
+
+      _ <- dateAccessUpdaterQueue.offer(UpdateDateAccessMessage(runtimeName, runtime.cloudContext, ctx.now))
```

Comment:
would be nice if we can refactor the msg to be an ADT, sth like
```
sealed abstract class UpdateDateAccessMessage extends Product with Serializable
object UpdateDateAccessMessage {
 final case class UpdateDateAccessMessageRuntimeV1(runtimeName: RuntimeName, cloudContext: CloudContext, dateAccessd: Instant) extends UpdateDateAccessMessage {
  override def toString: String =
    s"Message: ${cloudContext.asStringWithProvider}/${runtimeName.asString}, ${dateAccessd.toEpochMilli}"
}
 final case class UpdateDateAccessMessageRuntimeV2(runtimeName: RuntimeName, workspaceId: WorkspaceId, dateAccessd: Instant) extends UpdateDateAccessMessage {
  override def toString: String =
    s"Message: ${workspaceId}/${runtimeName.asString}, ${dateAccessd.toEpochMilli}"
}
}
```
Just so we can slowly move away from relying on `cloudContext`

Reply:
Lets chat in our 1-on-1 tomorrow on this!